libafl_qemu/src/modules/usermode/redirect_stdin.rs

```diff
@@ -0,0 +1,93 @@
+use libafl_qemu_sys::GuestAddr;
+
+use crate::{
+    Qemu, SYS_execve, SYS_read,
+    emu::EmulatorModules,
+    modules::{EmulatorModule, EmulatorModuleTuple},
+    qemu::{Hook, SyscallHookResult},
+};
+
+/// This module hijacks any read to buffer from stdin, and instead fill the buffer with the specified input address instead
+/// This is useful when your binary target reads the input from the stdin.
+/// With this you can just fuzz more like afl++
+/// You need to use this with snapshot module!
+#[derive(Debug, Default)]
+pub struct RedirectStdinModule {
```

Comment:
any suggestions for better name?

Reply:
StreamInputModule?

Also, why only stdin?

Reply:
well you don't really know which is the number of fd. if it is not about stdin.
also if it is a file then it is more complicated, for example, you have to take care of lseek, ..etc

Reply:
in any case, if it is just a file, you could just write input to the file before execution. in this situation, things are more simpler

Reply:
You could hook open() and assign a dummy fd, or just redirect all reads optionally

Reply:
so why not just mmap the file to memory in that case instead of doing that.
this module is meant to be a special handling for stdin where you can't really map stdin to memory.

Reply:
but i'm still trying to do this in a way that I wrote in discord. which would be generic for both stdin and file

Reply:
for files, ok maybe. You still need an extra round trip where you write the file data to the mapped "file", etc.

For stdin just reading from the input directly via unsafe is the right way to go for sure

Reply:
> For stdin just reading from the input directly via unsafe is the right way to go for sure

yes. i noticed later but i was actually doing this
see my comment on the other thread. it was not doing any extra copy at all in fact.